R/dash.R

```diff
@@ -257,17 +256,57 @@ Dash <- R6::R6Class(
         TRUE
       })
 
-      # TODO: once implemented in dash-renderer, leverage this endpoint so
-      # we can dynamically load dependencies during `_dash-update-component`
-      # https://plotly.slack.com/archives/D07PDTRK6/p1507657249000714?thread_ts=1505157408.000123&cid=D07PDTRK6
-      dash_suite <- paste0(routes_pathname_prefix, "_dash-component-suites")
+      # This endpoint supports dynamic dependency loading 
+      # during `_dash-update-component` -- for reference:
+      # https://github.com/plotly/dash/blob/1249ffbd051bfb5fdbe439612cbec7fa8fff5ab5/dash/dash.py#L488
+      # https://docs.python.org/3/library/pkgutil.html#pkgutil.get_data
+      dash_suite <- paste0(self$config$routes_pathname_prefix, "_dash-component-suites/:package_name/*")
       route$add_handler("get", dash_suite, function(request, response, keys, ...) {
-
-        response$status <- 500L
-        response$body <- "Not yet implemented"
+        # we're looking for a bit more information than returned by the routr API
+        # for the handler keys, so the URL parsing is done manually here:
+        #
+        # 1) extract the URL from the request
+        # 2) split the URL into chunks for each of the /*/ blocks
+        # 3) retrieve the script pathname by starting from the first block following
+        #    the "package" name (for the Dash deps, not the R package) name in the
+        #    URL, then capturing the rest of the pathname afterwards
+        # 4) concatenate the strings into a path, collapsing with /
+        # 5) pass the constructed path to get_package_mapping to retrieve R pkg name
+        url <- request$url
+        url_as_strings <- strsplit(gsub("http://|https://|www\\.", "", url), "/")[[1]]
```

Comment:
It might be cleaner to just use the path pattern matching itself. If you write your URL like:
```
       dash_suite <- paste0(self$config$routes_pathname_prefix, "_dash-component-suites/:package_name/:filename")
```

Then I believe that:
```
dist_path = keys$filename
dep_pkg = keys$package_name
```
will be what you're looking for.

Reply:
I tried to work with the information within the `keys` passed to the route, but it's not quite what we want. Using the URL syntax you provided above, I am able to :hocho: the messy `url` parsing block from L275:L278; thanks for noting this.

But `dep_pkg` doesn't line up with what is returned by `get_package_mapping`. Here are two dependencies as an example:

previous `dist_path`: `plotly-1.44.1.min.js`
new `dist_path`: `plotly-1.44.1.min.js`
previous `dep_pkg`: `deps/plotly-1.44.1.min.js`  
new `dep_pkg`: `dash_core_components_0` 

previous `dist_path`: `dash_core_components.min.js` 
new `dist_path`: `dash_core_components.min.js` 
previous `dep_pkg`: `deps/dash_core_components.min.js`
new `dep_pkg`: `dash_core_components`

Reply:
fixed in 167079a9b747926794625f8eb51d6bb01756adf2